### PR TITLE
Cleaned up StakeMiner.

### PR DESCRIPTION
**Pull request by tomasbrod**
* Removed dead code
* unused PoW functionality.
* Added miner status structure
* added miner status to getmininginfo rpc
* replaced getstakinginfio rpc with alias to getmininginfo
* Use MinerStatus in some GlobalStatus fields.
* Fixed display of Owed rewards in getmininginfo rpc.

Note that some of the data will be zero when not staking and will be less than true when kernel is found.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -97,7 +98,6 @@ extern std::string GetCurrentNeuralNetworkSupermajorityHash(double& out_populari
 extern std::string GetNeuralNetworkSupermajorityHash(double& out_popularity);
        
 extern double CalculatedMagnitude2(std::string cpid, int64_t locktime,bool bUseLederstrumpf);
-extern int64_t ComputeResearchAccrual(int64_t nTime, std::string cpid, std::string operation, CBlockIndex* pindexLast, bool bVerifyingBlock, int VerificationPhase, double& dAccrualAge, double& dMagnitudeUnit, double& AvgMagnitude);
```
Comment (denravonska) [inline src/main.cpp:100]: Thank you! The more forward declarations we remove the better.

File: src/miner.h [hunk 2]
```diff
@@ -9,22 +9,37 @@
...
+    double WeightSum,WeightMin,WeightMax;
+    double ValueSum;
+    double CoinAgeSum;
+    int Version;
+    unsigned long CreatedCnt;
+    unsigned long AcceptedCnt;
+    unsigned long KernelsFound;
+    int64_t nLastCoinStakeSearchInterval;
+
+    void Clear()
```
Comment (denravonska) [inline src/miner.h]: I prefer code in cpp files or we end up with main.h. Not a deal breaker for this PR though :)

Edit: I probably don't live as I learn and most likely did the same thing with the global status struct :(

File: src/rpcmining.cpp [hunk 3]
```diff
@@ -99,455 +125,7 @@ Value getmininginfo(const Array& params, bool fHelp)
...
-    result.push_back(Pair("sizelimit", (int64_t)MAX_BLOCK_SIZE));
-    result.push_back(Pair("curtime", (int64_t)pblock->nTime));
-    result.push_back(Pair("bits", strprintf("%08x", pblock->nBits)));
-    result.push_back(Pair("height", (int64_t)(pindexPrev->nHeight+1)));
-
-    return result;
+    return getmininginfo(params,fHelp);
 }
 
 Value submitblock(const Array& params, bool fHelp)
```
Comment (denravonska) [inline src/rpcmining.cpp]: `submitblock` doesn't seem to do anything as `ProcessBlock` is commented out. I opt for removing it as well.

**Comment by denravonska** [comment]
Tooooons of obsolete code removed. Gj!

File: src/miner.cpp [hunk 4]
```diff
@@ -29,46 +30,13 @@ StructCPID GetLifetimeCPID(const std::string& cpid, const std::string& sFrom);
...
-
-void SHA256Transform(void* pstate, void* pinput, const void* pinit)
-{
-    SHA256_CTX ctx;
-    unsigned char data[64];
-
-    SHA256_Init(&ctx);
-
-    for (int i = 0; i < 16; i++)
-        ((uint32_t*)data)[i] = ByteReverse(((uint32_t*)pinput)[i]);
```
Comment (denravonska) [inline src/miner.cpp:63]: `ByteReverse` in util.h is now unused and can be removed as well.

File: src/miner.h (hunk 2 above)
Comment (tomasbrod) [reply src/miner.h]: The problem with code in header is that it is implicitly inlined. Which adds compile time and duplicates code in resulting binary.

File: src/qt/bitcoingui.cpp [hunk 5]
```diff
@@ -2089,5 +2101,9 @@ void BitcoinGUI::updateStakingIcon()
             labelStakingIcon->setToolTip(tr("Not staking"));
 			msMiningErrors6 = "Not staking yet";
 		}
+	*/
+
+	//This string wont be translated :(
+	labelStakingIcon->setToolTip("Not staking; "+QString(ReasonNotStaking.c_str()));
```
Comment (denravonska) [inline src/qt/bitcoingui.cpp]: Missing tr macro.

**Review by denravonska — CHANGES_REQUESTED** [review]
Couple of small changes in comments.

**Pull request merged**